core/src/main/java/org/jdbi/v3/spi/JdbiPlugin.java

```diff
@@ -0,0 +1,20 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.spi;
+
+import org.jdbi.v3.Handle;
+
+public interface JdbiPlugin {
+    default Handle customizeHandle(Handle handle) { return handle; }
```

Comment:
I realize customizing a statement can be done by providing a wrapper handle, but that is kind of inconvenient compared to making it first class on the plugin. It also implies separate methods for customizing queries, statements, batches, etc, so as to handle all the downcast cases from statement correctly.